libraries/cmd/armed.cpp

```diff
@@ -198,10 +198,14 @@ void Armed::ActivateGuns(const WeaponInfo *sz, bool ms) {
 }
 
 void Armed::Fire(unsigned int weapon_type_bitmask, bool listen_to_owner) {
-    Unit *unit = static_cast<Unit *>(this);
+    Unit *unit = vega_dynamic_cast_ptr<Unit>(this);
 
-    if ((unit->cloak.Active() && !configuration()->weapons.can_fire_in_cloak) ||
-            (unit->ftl_drive.Enabled() && !configuration()->weapons.can_fire_in_spec)) {
+    if (unit->cloak.Active() && !configuration()->weapons.can_fire_in_cloak) {
+        VS_LOG(debug, (boost::format("%1%: can't fire while cloaked") % __FUNCTION__));
```

Comment:
These will potentially spam the logs.

Reply:
Yeah, they might. I will probably take these out before I finalize this PR.

Reply:
I believe I changed these to `trace` level as well. (In the other PR)